File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    dict_ = {}
    answer = 0
    
    for i in array:
        if i in dict_:
            dict_[i] += 1
        else:
            dict_[i] = 1

    for i in dict_:
        if dict_[i] == max(dict_.values()):
            if answer != 0:
                return -1
            answer = i
    return answer

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        Map<Integer, Integer> map = new HashMap<>();
        int maxCount = 0;
        int answer = 0;
        
        for (int i : array) {
            if (map.containsKey(i)) {
                map.put(i, map.get(i) + 1);
            } else {
                map.put(i, 1);
            }
            
            int count = map.get(i);
            if (count > maxCount) {
                maxCount = count;
                answer = i;
            } else if (count == maxCount) {
                answer = -1; 
            }
        }
        
        return answer;
    }
}
